components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -25,6 +25,68 @@ const getWindowId = (id: number) => {
   return `id_${id}`
 }
 
+function pollTwitchPage (tab: chrome.tabs.Tab, tabId: number, publisherBlob:string): void {
+  let url = tab.url
+  // use an interval here to monitor when the DOM has finished
+  // generating. clear after the data is present.
```

Comment:
why not just use 

```chrome.tabs.onUpdated.addListener(function (tabId , info) {
  if (info.status === 'complete') {
    // your code ...
  }
});
```
instead of polling?

Reply:
For twitch pages, the DOM is loaded dynamically and is not guaranteed to have the data we need at 'complete'